shell/platform/linux/fl_key_channel_responder_test.cc

```diff
@@ -0,0 +1,204 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_key_channel_responder.h"
+
+#include "gtest/gtest.h"
+
+#include "flutter/shell/platform/linux/fl_binary_messenger_private.h"
+#include "flutter/shell/platform/linux/fl_engine_private.h"
+#include "flutter/shell/platform/linux/testing/fl_test.h"
+
+static const char* expected_value = nullptr;
+static gboolean expected_handled = FALSE;
+
+static FlValue* echo_response_cb(FlValue* echoed_value) {
+  gchar* text = fl_value_to_string(echoed_value);
+  EXPECT_STREQ(text, expected_value);
+  g_free(text);
+
+  FlValue* value = fl_value_new_map();
+  fl_value_set_string_take(value, "handled",
+                           fl_value_new_bool(expected_handled));
+  return value;
+}
+
+static void responder_callback(bool handled, gpointer user_data) {
+  EXPECT_EQ(handled, expected_handled);
+  g_main_loop_quit(static_cast<GMainLoop*>(user_data));
+}
+
+namespace {
+// A dummy object, whose pointer will be used as _g_key_event->origin.
+static int _origin_event;
+// A global variable to store new event. It is a global variable so that it can
+// be returned by key_event_new for easy use.
+static FlKeyEvent _g_key_event;
+}  // namespace
+
+// Clone #string onto the heap.
```

Comment:
Don't add markup in implementation comments.
```suggestion
// Clone string onto the heap.
```